Enable Ninja Builds in Visual Studio (#34851)
mantidproject/mantid: dev-docs/source/GettingStarted/GettingStartedCondaWindows.rst

Configure CMake and generate build files
----------------------------------------

* You can configure CMake using an MS Visual Studio or Ninja generator. Choose one of the following:
    * For MS Visual Studio, use the terminal or powershell prompt from the last step.
    * For Ninja, open the ``x64 Native Tools Command Prompt for VS 2019`` or ``x64 Native Tools Command Prompt for VS 2022`` from your search bar.

* Navigate to your mantid source directory.
* Run ``conda activate mantid-developer`` to activate your conda environment.
* If you want your build directory inside your source directory, run either:

    * ``cmake --preset=win-vs`` for configuring with Visual Studio 2022, or
    * ``cmake --preset=win-vs-2019`` for configuring with Visual Studio 2019, or
    * ``cmake --preset=win-ninja`` for configuring with Ninja.

* Alternatively, if you want to specify a different build directory, run either:
    * ``cmake --preset=win-vs -B {BUILD_DIR}``, or
    * ``cmake --preset=win-vs-2019  -B {BUILD_DIR}``, or
    * ``cmake --preset=win-ninja -B {BUILD_DIR}``

Compile and Build using MS Visual Studio
----------------------------------------

> * Open visual studio with ``visual-studio.bat`` then click build.

Reviewer: ```suggestion
* Open visual studio with ``visual-studio.bat``, which is found in the build folder, and then click build.
```

* It's not possible to compile in Debug on Windows with conda libraries, however Release, RelWithDebInfo, and DebugWithRelRuntime for Debugging will compile fine.
* Once in visual studio, the correct target to use as a startup project in visual studio is ``workbench``, not ``MantidWorkbench``. You can then press F5 to start workbench.

Compile and Build using Ninja
-----------------------------
* From the command line:
    * Navigate to the build directory using the ``x64 Native Tools Command Prompt for VS 2019`` or ``x64 Native Tools Command Prompt for VS 2022`` from the previous step.
    * To build Mantid Workbench use: ``ninja``
    * To build the unit tests use: ``ninja AllTests``

* In Visual Studio:

>     * Open Visual Studio using ``visual-studio_ninja.bat``

Reviewer: ```suggestion
    * Open Visual Studio with ``visual-studio_ninja.bat``, which is found in the build folder
```

    * Select "win-ninja" from the Configuration dropdown at the top of the screen

>     * Select "Switch between solutions and available views" in the Solution Explorer and click on CMake Targets View

Reviewer: ```suggestion
    * Select "Switch between solutions and available views" in the Solution Explorer and click on "CMake Targets View"
```

>     * It is recommmended that the option "When cache is out of date" option is set to "Never run configure step automatically" in Tools, Options, CMake

Reviewer: ```suggestion
    * It is recommmended that the "When cache is out of date" option is set to "Never run configure step automatically" in Tools->Options->CMake
```
